website/docs/r/s3_bucket_policy.html.markdown

```diff
@@ -53,7 +50,7 @@ resource "aws_s3_bucket_policy" "b" {
 The following arguments are supported:
 
 * `bucket` - (Required) The name of the bucket to which to apply the policy.
-* `policy` - (Required) The text of the policy. For more information about building AWS IAM policy documents with Terraform, see the [AWS IAM Policy Document Guide](https://learn.hashicorp.com/terraform/aws/iam-policy). Note: Bucket policies are limited to 20 KB in size.
+* `policy` - (Required) The text of the policy. Although this isn't technically an IAM policy, you can use the [`aws_iam_policy_document`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) data source for this, so long as it specifies a principal. For more information about building AWS IAM policy documents with Terraform, see the [AWS IAM Policy Document Guide](https://learn.hashicorp.com/terraform/aws/iam-policy). Note: Bucket policies are limited to 20 KB in size.
```

Comment:
Since the note you're adding is covered in the linked AWS IAM policy document guide, I don't believe it's necessary to add it here as well.

Reply:
i'm a little conflicted on this - i modeled this message after the one on [`aws_kms_key`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/kms_key), though with slightly friendlier language. the other thing is that the tutorial, while helpful in learning best practice for IAM policies, isn't necessarily the best way of consuming quick-reference information.

i'm happy to make the changes you're suggesting obviously - you're the maintainer here 😅 but thought it might be a good idea to detail the reasoning behind why i wrote what i did

Reply:
Thanks for the link! I see where you're coming from now (and wasn't really ever _against_ the idea, for what it's worth 😃). With that in mind, I think this rephrasing might present a bit clearer.

```suggestion
* `policy` - (Required) The text of the policy. Although this is a bucket policy rather than an IAM policy, the [`aws_iam_policy_document`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) data source may be used, so long as it specifies a principal. For more information about building AWS IAM policy documents with Terraform, see the [AWS IAM Policy Document Guide](https://learn.hashicorp.com/terraform/aws/iam-policy). Note: Bucket policies are limited to 20 KB in size.
```

Reply:
good suggestion 👍 have applied